mapiproxy/servers/default/emsmdb/oxcfxics.c

```diff
@@ -3480,27 +3545,58 @@ static enum MAPISTATUS oxcfxics_fill_transfer_state_arrays(TALLOC_CTX *mem_ctx,
 						  emsmdbp_get_contextID(table_object), table_object->backend_object,
 						  synccontext->properties.cValues, synccontext->properties.aulPropTag);
 		OPENCHANGE_RETVAL_IF(ret != MAPISTORE_SUCCESS, mapistore_error_to_mapi(ret), local_mem_ctx);
+		if (synccontext->request.is_collector) {
+			/* Set restriction on which change numbers and
+			   fmid we are interested in */
+			ret = oxcfxics_transfer_state_set_table_restriction(emsmdbp_ctx,
+									    table_object,
+									    synccontext);
+			OPENCHANGE_RETVAL_IF(ret != MAPISTORE_SUCCESS, mapistore_error_to_mapi(ret), local_mem_ctx);
+
+			ret = mapistore_table_get_row_count(emsmdbp_ctx->mstore_ctx, emsmdbp_get_contextID(table_object),
+							    table_object->backend_object, MAPISTORE_PREFILTERED_QUERY,
+							    &table_object->object.table->denominator);
+			OPENCHANGE_RETVAL_IF(ret != MAPISTORE_SUCCESS, mapistore_error_to_mapi(ret), local_mem_ctx);
+
+			OC_DEBUG(5, "Number of updated messages to return: %d with CN >= 0x%.16"PRIx64,
+                                 table_object->object.table->denominator, synccontext->next_cn);
+
+			OPENCHANGE_RETVAL_IF(table_object->object.table->denominator == 0,
+					     MAPI_E_SUCCESS, local_mem_ctx);
+		}
 	}
+
+	if (synccontext->request.is_collector) {
+		involved_fmids = RAWIDSET_convert_to_idset(local_mem_ctx, synccontext->involved_fmids);
```

Comment:
I'm against that logic. Indeed, I haven't changed previous logic because I didn't want to make this kind of changes without a full wipe-out. What's the benefit of freeing here the idsets? IMHO, talloc local contexts do the job perfectly to dynamic allocate variables for a function at the beginning and, then, at the end freeing them. Am I nuts?